@here/harp-map-controls/lib/CameraKeyTrackAnimation.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { MapView, MapViewEventNames, MapViewUtils } from "@here/harp-mapview";
+import { LoggerManager } from "@here/harp-utils";
+import THREE = require("three");
+
+import { CameraAnimation } from "./MapAnimations";
+import { CameraPose, CameraPoseOptions } from "./Utils";
+
+const logger = LoggerManager.instance.create("CameraKeyTrackAnimation");
+
+export interface ControlPointOptions extends CameraPoseOptions {
+    timestamp: number;
+}
+/**
+ * A ControlPoint used to defined the Track of the [[CameraKeyTrackAnimation]]
+ */
+export class ControlPoint extends CameraPose {
+    timestamp: number = 0;
+    constructor(options: ControlPointOptions) {
+        super(options);
+        this.timestamp = options.timestamp;
+    }
+}
+
+export interface CameraKeyTrackAnimationOptions {
+    controlPoints: ControlPoint[];
+    repeat?: boolean;
+    name?: string;
+    interpolation?: THREE.InterpolationModes;
+    loop?: THREE.AnimationActionLoopStyles; //types seem outdate too, should be THREE.LoopModes
+    rotateOnlyClockwise?: boolean;
+    repetitions?: number;
+}
+
+class AnimationDummy extends THREE.Object3D {
+    distance: number = 0;
+
+    constructor(name: string) {
+        super();
+        this.name = name;
+    }
+}
+
+/**
+ * A [[CameraAnimation]] along of a set of [[ControlPoint]]s
+ */
+export class CameraKeyTrackAnimation extends CameraAnimation {
```

Comment:
What is benefit of inheriting from this class ? It completly ignores original implementation and bring unnecessary code as dependency.

If interfacing with old animations and new animations in code is important consider creation of ICameraAnimation interface.
This for example brings whole tweenjs

Reply:
It actually implements the parts of the abstract class CameraAnimation, although the usage is minor, it still applies.But good point, that the CameraAnimation is actually carved for the usage with tween. I ll remove the dependency.